core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-block-editor-sidebar/dot-block-editor-sidebar.component.ts

```diff
@@ -0,0 +1,134 @@
+import { DotEventsService } from '@services/dot-events/dot-events.service';
+import { Observable, of, Subject } from 'rxjs';
+import { DotBlockEditorComponent } from '@dotcms/block-editor';
+import { HttpErrorResponse } from '@angular/common/http';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotGlobalMessageService } from '@components/_common/dot-global-message/dot-global-message.service';
+import { DotContentTypeService } from '@services/dot-content-type';
+import { DotCMSContentTypeField, DotCMSContentTypeFieldVariable } from '@dotcms/dotcms-models';
+
+export interface BlockEditorInput {
+    content: { [key: string]: string };
+    fieldName: string;
+    language: number;
+    inode: string;
+    fieldVariables?: {
+        allowedBlocks?: string;
+        allowedContentTypes?: string;
+        styles: string;
+    };
+}
+
+@Component({
+    selector: 'dot-block-editor-sidebar',
+    templateUrl: './dot-block-editor-sidebar.component.html',
+    styleUrls: ['./dot-block-editor-sidebar.component.scss']
+})
+export class DotBlockEditorSidebarComponent implements OnInit, OnDestroy {
+    @ViewChild('blockEditor') blockEditor: DotBlockEditorComponent;
+
+    data: BlockEditorInput;
+    saving = false;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private dotWorkflowActionsFireService: DotWorkflowActionsFireService,
+        private dotEventsService: DotEventsService,
+        private dotMessageService: DotMessageService,
+        private dotGlobalMessageService: DotGlobalMessageService,
+        private dotContentTypeService: DotContentTypeService
+    ) {}
+
+    ngOnInit(): void {
+        this.dotEventsService
+            .listen<HTMLDivElement>('edit-block-editor')
+            .pipe(
+                takeUntil(this.destroy$),
+                switchMap((event) => this.extractBlockEditorData(event.data.dataset))
+            )
+            .subscribe((eventData: BlockEditorInput) => {
+                this.data = eventData;
+            });
+    }
+
+    /**
+     *  Execute the workflow to save the editor changes and then close the sidebar.
+     *
+     * @memberof DotBlockEditorSidebarComponent
+     */
+    saveEditorChanges(): void {
+        this.saving = true;
+        this.dotWorkflowActionsFireService
+            .saveContentlet({
+                [this.data.fieldName]: JSON.stringify(this.blockEditor.editor.getJSON()),
+                inode: this.data.inode,
+                indexPolicy: 'WAIT_FOR'
+            })
+            .pipe(take(1))
+            .subscribe(
+                () => {
+                    this.saving = false;
+                    const customEvent = new CustomEvent('ng-event', {
+                        detail: { name: 'in-iframe' }
+                    });
+                    window.top.document.dispatchEvent(customEvent);
+                    this.data = null;
```

Comment:
If this is for close the dialog... should we call `closeSidebar`?

Reply:
right !